osu.Game/Overlays/NowPlayingOverlay.cs

```diff
@@ -469,5 +471,98 @@ protected override void OnHoverLost(HoverLostEvent e)
                 base.OnHoverLost(e);
             }
         }
+
+        private partial class ScrollingTextContainer : CompositeDrawable
+        {
+            private const float initial_move_delay = 1000;
+            private const float pixels_per_second = 50;
+
+            private LocalisableString text;
+            private OsuSpriteText mainSpriteText = null!;
+            private OsuSpriteText fillerSpriteText = null!;
+
+            public LocalisableString Text
+            {
+                get => text;
+                set
+                {
+                    text = value;
+                    Schedule(updateText);
+                }
+            }
+
+            public FontUsage Font
+            {
+                set =>
+                    Schedule(() =>
+                    {
+                        mainSpriteText.Font = value;
+                        fillerSpriteText.Font = value;
+
+                        updateText();
+                    });
+            }
+
+            public ScrollingTextContainer()
+            {
+                AutoSizeAxes = Axes.Both;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                InternalChild = new FillFlowContainer<OsuSpriteText>
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Direction = FillDirection.Horizontal,
+                    Children = new[]
+                    {
+                        mainSpriteText = new OsuSpriteText { Padding = new MarginPadding { Horizontal = margin } },
+                        fillerSpriteText = new OsuSpriteText { Padding = new MarginPadding { Horizontal = margin }, Alpha = 0 },
+                    }
+                };
+            }
+
+            private void updateText()
+            {
+                mainSpriteText.Text = text;
+                fillerSpriteText.Alpha = 0;
+
+                ClearTransforms();
+                X = 0;
+
+                float textOverflowWidth = mainSpriteText.Width - player_width;
+
+                if (textOverflowWidth > 0)
```

Comment:
I think this needs at least a little bit of tolerance because looking at the test scene there is no real actual reason why the artist should be scrolling here:

![1714370676](https://github.com/ppy/osu/assets/20418176/f8b9a200-28d9-4d02-af74-e98f0f5959eb)

the artist name width exceeds the width of the entire overlay by like .08005 units or something.